src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegion.java

```diff
@@ -0,0 +1,609 @@
+            throw new IllegalArgumentException("Read location " + readLoc + " doesn't overlap with active region extended span " + extendedLoc);
+        }
+
+        spanIncludingReads = spanIncludingReads.mergeWithContiguous( readLoc );
+
+        if ( ! reads.isEmpty() ) {
+            final GATKRead lastRead = reads.get(size() - 1);
+            if ( ! Objects.equals(lastRead.getContig(), read.getContig()) ) {
+                throw new IllegalArgumentException("Attempting to add a read to ActiveRegion not on the same contig as other reads: lastRead " + lastRead + " attempting to add " + read);
+            }
+            if ( read.getStart() < lastRead.getStart() ) {
+                throw new IllegalArgumentException("Attempting to add a read to ActiveRegion out of order w.r.t. other reads: lastRead " + lastRead + " at " + lastRead.getStart() + " attempting to add " + read + " at " + read.getStart());
+            }
+        }
+
+        reads.add( read );
+    }
+
+    /**
+     * Get the number of reads currently in this region
+     * @return an integer >= 0
+     */
+    public int size() { return reads.size(); }
+
+    /**
+     * Clear all of the reads currently in this region
+     */
+    public void clearReads() {
+        spanIncludingReads = extendedLoc;
+        reads.clear();
+    }
+
+    /**
+     * Remove all of the reads in readsToRemove from this region
+     * @param readsToRemove the set of reads we want to remove
+     */
+    public void removeAll( final Collection<GATKRead> readsToRemove ) {
+        Utils.nonNull(readsToRemove);
+        reads.removeAll(readsToRemove);
+        spanIncludingReads = extendedLoc;
+        for (final GATKRead read : reads) {
+            spanIncludingReads = spanIncludingReads.mergeWithContiguous(read);
+        }
+    }
+
+    /**
+     * Add all readsToAdd to this region
+     * @param readsToAdd a collection of readsToAdd to add to this active region
+     */
+    public void addAll(final Collection<GATKRead> readsToAdd){
+        Utils.nonNull(readsToAdd).forEach(r -> add(r));
+    }
+
+    /**
+     * Get the extension applied to this region
+     *
+     * The extension is >= 0 bp in size, and indicates how much padding this art walker wanted for its regions
+     *
+     * @return the size in bp of the region extension
+     */
+    public int getExtension() { return extension; }
+
+    public SimpleInterval getReadSpanLoc() {
+        return spanIncludingReads;
+    }
+
+    /**
+     * An ordered list (by genomic coordinate) of the ActivityProfileStates that went
+     * into this active region.  May be empty, which says that no supporting states were
+     * provided when this region was created.
+     * The returned list is unmodifiable.
+     */
+    public List<ActivityProfileState> getSupportingStates() {
+        return supportingStates;
+    }
+
+    /**
+     * See #getActiveRegionReference but using the span including regions not the extended span
+     */
```

Comment:
I think this "live array" comment was incorrect -- the `referenceReader.getSubsequenceAt()` call actually allocates a new byte array. Removing this warning.